File: IonicResearch.Data/Models/OutletForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace IonicResearch.Data.Models
{
    public class OutletForm
    {
        public int Id { get; set; }
        public DateTime RepDate { get; set; }
        [MaxLength(200)]
        public string Name { get; set; }
        [MaxLength(200)]
        public string Inn { get; set; }
        [MaxLength(200)]
        public string Address { get; set; }
        public long OpeningTime { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
        public DateTime CreatedDate { get; set; }
        public int UserId { get; set; }
        public int OutletTypeId { get; set; }

        public User User { get; set; }
        public OutletType Type { get; set; }
        public List<OutletFormProduct> FormProducts { get; set; }
        public List<OutletPhoto> Photos { get; set; }
    }
}

File: IonicResearch.SPA/src/app/models/outlet-form.ts
import { OutletPhoto } from './outlet-photo';
import { OutletProduct } from './outlet-product';

export class OutletForm {
  id: number;
  repDate: Date;
  name: string;
  inn: string;
  address: string;
  openingTime: Date;
  latitude: number;
  longitude: number;
  typeId: number;
  photos: OutletPhoto[];
  products: OutletProduct[];

  constructor() {
    this.id = 0;
    this.products = [];
    this.photos = [];
  }
}
